### PR TITLE
chore: clickhouse v1.0 upgrade

### PR DESCRIPTION
**Pull request by loomts**

### CODE REVIEW

**Review by shanshanying — COMMENTED** [review]
and update `app.kubernetes.io/name=clickhouse-24` to `apps.kubeblocks.io/component-name=clickhouse` in Notes.text.

**Review by sophon-zt — DISMISSED** [review]

File: addons/clickhouse/templates/_helpers.tpl [hunk 1]
```diff
@@ -49,3 +49,95 @@ Selector labels
...
+{{- define "clickhouse24.cmpdRegexpPattern" -}}
+^clickhouse-24.*
+{{- end -}}
+
+{{/*
+Define ch-keeper24 component definition name
+*/}}
+{{- define "ch-keeper24.cmpdName" -}}
+{{- if eq (len .Values.cmpdVersionPrefix.keeper24 ) 0 -}}
+ch-keeper-24-{{ .Chart.Version }}
```
Comment (Y-Rookie) [inline addons/clickhouse/templates/_helpers.tpl:83]: `ch-keeper-24-XX` ->  `clickhouse-keeper-24` ，use fullname would be better

File: addons/clickhouse/templates/cmpd-clickhouse.yaml [hunk 2]
```diff
@@ -128,7 +133,7 @@ spec:
       valueFrom:
         componentVarRef:
           compDef: ch-keeper-24
```
Comment (Y-Rookie) [inline addons/clickhouse/templates/cmpd-clickhouse.yaml:135]: check if compDef here should use regex pattern match

File: addons/clickhouse/templates/clusterdefinition.yaml [hunk 3]
```diff
@@ -4,24 +4,18 @@ metadata:
...
-  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
-  {{- end }}
 spec:
   topologies:
     - name: standalone
       components:
         - name: clickhouse
-          compDef: clickhouse-24
+          compDef: {{ include "clickhouse24.cmpdRegexpPattern" . }}
     - name: clickhouse-cluster
```
[the remark this answers is not in the document]
Comment (loomts) [reply addons/clickhouse/templates/clusterdefinition.yaml]: fixed

File: addons/clickhouse/templates/cmpd-clickhouse.yaml (hunk 2 above)
Comment (loomts) [reply addons/clickhouse/templates/cmpd-clickhouse.yaml]: fixed

**Review by Y-Rookie — APPROVED** [review]

**Review by Y-Rookie — APPROVED** [review]

**Review by sophon-zt — APPROVED** [review]

**Pull request merged**